### PR TITLE
Add Tests

### PR DESCRIPTION
**Pull request by glushkovaWT**

### CODE REVIEW

File: src/main/java/com/example/gymbuddy/implementation/configurations/ModelMapperConfig.java [hunk 1]
```diff
@@ -1,13 +1,29 @@
...
+
 @Configuration
 public class ModelMapperConfig {
     @Bean
     public ModelMapper modelMapper() {
+//        var modelMapper = new ModelMapper();
+//        modelMapper.
         return new ModelMapper();
     }
+
```
Comment (lucas-buccilli) [inline src/main/java/com/example/gymbuddy/implementation/configurations/ModelMapperConfig.java:20]: Remove commented out code

File: src/main/java/com/example/gymbuddy/implementation/services/MachineService.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    @Override
+    public List<MachineDto> findAll() {
+        return machineDataProvider.findAll();
+    }
+
+    @Override
+    public MachineDto addMachine(MachineDto machineDto) {
+        var exists = machineDataProvider.findByName(machineDto.getName()).isPresent();
+        if(exists) throw new RuntimeException("Exists");
```
Comment (lucas-buccilli) [inline src/main/java/com/example/gymbuddy/implementation/services/MachineService.java:25]: Create a custom exception like DuplicateEntityException

File: src/test/java/com/example/gymbuddy/implementation/controllers/MachineHistoryControllerTest.java [hunk 3]
```diff
@@ -48,18 +62,84 @@ void addMachineHistory() throws Exception {
...
+    @Test
+    public void getLatestMachineHistoryTrowsException() throws Exception {
+
+        //what is the exception? 404?
+        when(machineHistoryService.findLatestWorkout(anyInt(), anyInt()))
+                .thenReturn(Optional.empty());
+
+        mockMvc.perform(get("/members/1/machines/2/history/latest"))
+                .andExpect(status().isNotFound());
+                //.andExpect(jsonPath("$.timestamp").isNotEmpty())
```
Comment (lucas-buccilli) [inline src/test/java/com/example/gymbuddy/implementation/controllers/MachineHistoryControllerTest.java:140]: Remove commented out code

File: src/test/java/com/example/gymbuddy/integration/repository/MachineHistoryRepositoryIntegrationTest.java [hunk 4]
```diff
@@ -56,8 +57,41 @@ void temp() {
...
         assertEquals(0, machineHistories.size());
 
-        specification = MachineHistorySpecificationBuilder.builder().hasWorkoutDate(LocalDate.now().minusDays(1L)).build();
+        specification = MachineHistorySpecificationBuilder.builder().hasWorkoutDate(LocalDateTime.now().minusDays(1L)).build();
         machineHistories = machineHistoryRepository.findAll(specification);
         assertEquals(0, machineHistories.size());
     }
+
+    @Test
+    void temp2() {
```
Comment (lucas-buccilli) [inline src/test/java/com/example/gymbuddy/integration/repository/MachineHistoryRepositoryIntegrationTest.java:66]: Name tests in this integration test to something more specific

**Review by lucas-buccilli — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/gymbuddy/implementation/configurations/ModelMapperConfig.java [hunk 5]
```diff
@@ -1,13 +1,29 @@
...
 
+import java.sql.Date;
+import java.time.LocalDateTime;
+
 @Configuration
 public class ModelMapperConfig {
     @Bean
     public ModelMapper modelMapper() {
+//        var modelMapper = new ModelMapper();
+//        modelMapper.
```
Comment (lucas-buccilli) [inline src/main/java/com/example/gymbuddy/implementation/configurations/ModelMapperConfig.java:17]: Remove commented out code

Comment (glushkovaWT) [reply src/main/java/com/example/gymbuddy/implementation/configurations/ModelMapperConfig.java:17]: done

**Review by lucas-buccilli — APPROVED** [review]

**Pull request merged**